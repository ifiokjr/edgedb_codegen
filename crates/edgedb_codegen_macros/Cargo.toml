[package]
name = "edgedb_codegen_macros"
version = { workspace = true }
authors = { workspace = true }
categories = ["database"]
edition = { workspace = true }
homepage = { workspace = true }
keywords = ["edgedb", "database", "typesafe", "checked", "macros"]
license = { workspace = true }
readme = "readme.md"
repository = { workspace = true }
rust-version = { workspace = true }
description = "The proc-macro crate for the `edgedb_codegen` crate."

[lib]
proc-macro = true

[dependencies]
edgedb_codegen_core = { workspace = true }
proc-macro2 = { workspace = true }
quote = { workspace = true }
syn = { workspace = true }

[features]
with_bigint = ["edgedb_codegen_core/with_bigint"]
with_bigdecimal = ["edgedb_codegen_core/with_bigdecimal"]
with_chrono = ["edgedb_codegen_core/with_chrono"]
with_all = ["edgedb_codegen_core/with_all"]
query = []
serde = []

[lints]
workspace = true
