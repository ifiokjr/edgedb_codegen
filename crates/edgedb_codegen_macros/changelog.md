# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.2.0](https://github.com/ifiokjr/edgedb_codegen/compare/edgedb_codegen_macros@v0.1.2...edgedb_codegen_macros@v0.2.0) - 2024-08-28

### <!-- 0 -->🎉 Added
- [**breaking**] add optional `builder` feature limited to `Input`

### <!-- 3 -->📚 Documentation
- reorder package fields

## [0.1.1](https://github.com/ifiokjr/edgedb_codegen/compare/edgedb_codegen_macros@0.1.0...edgedb_codegen_macros@0.1.1) - 2024-08-26

### <!-- 7 -->⚙️ Miscellaneous Tasks
- improve changelog generation

## [0.1.0](https://github.com/ifiokjr/edgedb_codegen/releases/tag/edgedb_codegen_macros-v0.1.0) - 2024-08-25

### 🎉 Added

- Initial release
