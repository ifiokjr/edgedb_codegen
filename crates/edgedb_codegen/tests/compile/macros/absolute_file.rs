use edgedb_codegen_macros::edgedb_query_raw;

fn main() {
	edgedb_query_raw!(insert_user, file: "/absolute/path/to/queries/insert_user.edgeql");
}
