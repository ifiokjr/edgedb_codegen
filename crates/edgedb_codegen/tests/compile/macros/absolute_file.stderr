error: absolute paths will only work on the current machine
 --> tests/compile/macros/absolute_file.rs:4:39
  |
4 |     edgedb_query_raw!(insert_user, file: "/absolute/path/to/queries/insert_user.edgeql");
  |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
